src/python/pants/engine/process_test.py

```diff
@@ -10,18 +10,29 @@
 from pants.engine.fs import CreateDigest, Digest, DigestContents, FileContent, PathGlobs, Snapshot
 from pants.engine.internals.scheduler import ExecutionError
 from pants.engine.process import (
+    BinaryPathRequest,
+    BinaryPaths,
     FallibleProcessResult,
     InteractiveProcess,
     Process,
     ProcessExecutionFailure,
     ProcessResult,
 )
 from pants.engine.rules import Get, rule
-from pants.testutil.rule_runner import QueryRule
+from pants.testutil.pants_integration_test import setup_tmpdir
+from pants.testutil.rule_runner import QueryRule, RuleRunner
 from pants.testutil.test_base import TestBase
 from pants.util.contextutil import temporary_dir
 
 
+def process_rule_runner() -> RuleRunner:
```

Comment:
This should be annotated with `@pytest.fixture` to ensure that each unit test gets a fresh instance. Also we've been calling it `rule_runner` for consistency.

Reply:
Alternatively, if we expect multiple rule runners in the file with different setups, you can call it `find_binary_rule_runner`.

Reply:
So, I'm not a huge fan of the fixtures. I don't see a great reason to do:
```
def test_my_stuff(rule_runner: RuleRunner) -> None:
    ..
```
rather than:
```
def test_my_stuff() -> None:
    rule_runner = my_rule_runner()
```

Reply:
I think it's totally valid to create a `RuleRunner()` inline. I do that in some places. But imo, it should either be inline, or it should be a fixture. Right now is a mix of having the function declared as top-level.

> So, I'm not a huge fan of the fixtures. I don't see a great reason 

Conformity with the greater Python ecosystem, which makes heavy use of fixtures. Our Plugin API is now aligned with the conventions.

We also use pre-built fixtures in other places too now like `caplog`, and I think it's good to be consistent with those.

Finally, it's important that plugin authors don't try using one rule runner for multiple tests, such as a class var or global consant. The fixture is meant to emphasize that you want a new instance every test.

Reply:
> The fixture is meant to emphasize that you want a new instance every test.

Unless you accidentally put `@pytest.fixture(scope=module)` on it, which would get you a reused one.

I don't feel strongly about it. But particularly if we're discouraging the reuse of fixtures, they don't seem to buy much.

Reply:
Ditto, I don't care that much either way, so long as we're consistent in plugin docs and we feel confident that users won't accidentally create one instance for >1 test.

I'm curious what John and Benjy think.

Reply:
I switched it so that we're consistent for now.